### PR TITLE
Add components Phase One

### PR DESCRIPTION
**Pull request by akshaaatt**
- Moved to bootstrap 5
- Added Assets
- Added Custom styles
- Header and Footer shared components added to storybook

### CODE REVIEW

File: scss/bootstrap-prod.scss [hunk 1]
```diff
@@ -5,45 +5,9 @@
  */
 
 .bs {
+    @import "../node_modules/bootstrap/scss/bootstrap";
```
Comment (akshaaatt) [inline scss/bootstrap-prod.scss:8]: @mwiencek Is this alright? I reckon we would want to ship bootstrap to all our projects from here directly and hence decided to do this. Let me know if this seems alright to you.

File: stories/index.stories.js [hunk 2]
```diff
@@ -238,40 +239,21 @@ storiesOf('Modal', module)
...
-    </Navbar>
+    <Header
+      projectName="musicbrainz"
+      searchOptions={searchOptions}
+    />
+  ));
+
+storiesOf('Footer', module)
+  .add('Default', () => (
+    <Footer
```
Comment (akshaaatt) [inline stories/index.stories.js:250]: Looking to do this way for all the projects and reduce boilerplate code. @mwiencek @MonkeyDo

Comment (mwiencek) [reply stories/index.stories.js:250]: We can't really include components in this repo that use translated strings, since we have no way of extracting them here (each project will have different i18n setups).  At best we could pass the strings in as props, but that would probably add more boilerplate than you're saving.

File: components/shared/Header.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                    English
+                  </a>
+                  <a className="dropdown-item ">
+                    Français
+                  </a>
+                  <a className="dropdown-item ">
+                    Italiano
+                  </a>
+                  <a className="dropdown-item ">
+                    Nederlands
```
Comment (mwiencek) [inline components/shared/Header.js:56]: The list of languages here are dynamic (configured in DBDefs.pm for MB) and wouldn't apply to projects outside of MB.  Given all the translatable strings in this file and project-specific implementation details, I don't see how we can have a shared component this specific -- it'd have to be much more generalized.

File: package.json [hunk 4]
```diff
@@ -42,7 +42,6 @@
     "eslint-plugin-react": "7.28.0",
     "eslint-plugin-storybook": "^0.5.6",
     "file-loader": "^6.2.0",
-    "popper.js": "^1.16.1",
```
Comment (mwiencek) [inline package.json:45]: The commit message doesn't explain why this is being removed.  I'm sure it's obvious to you, but it would be helpful to reviewers to explain it in the commit. :)

File: components/index.js [hunk 5]
```diff
@@ -1,3 +1,5 @@
+import 'bootstrap/dist/css/bootstrap.min.css';
```
Comment (mwiencek) [inline components/index.js]: I'm not sure how this change is related to "Remove popper js."  Should this be in a different commit?

File: package.json [hunk 6]
```diff
@@ -53,6 +57,7 @@
     "sass": "^1.49.0",
     "sass-loader": "^7.1.0",
     "style-loader": "^0.21.0",
+    "typescript": "^4.5.5",
```
Comment (mwiencek) [inline package.json:60]: Where do you use TypeScript in this PR?

Note that MB uses Flow, not TypeScript.  We could bundle both TS and Flow types in this repo in the future, but TS should probably be a dev or peer dependency instead.

Comment (akshaaatt) [reply package.json]: Removed typescript. Flow seems good for our usage!

File: components/index.js (hunk 5 above)
Comment (akshaaatt) [reply components/index.js]: Right!

File: package.json (hunk 4 above)
Comment (akshaaatt) [reply package.json:45]: https://getbootstrap.com/docs/5.0/migration/ Bootstrap now uses Popper v2.x. Hence, this dependency was causing conflicts.

File: scss/bootstrap-prod.scss (hunk 1 above)
Comment (akshaaatt) [reply scss/bootstrap-prod.scss:8]: I agree we should look to minimize the size of the CSS. I will look to import only useful individual modules now!
Also, I would like us to maintain the respective CSS here and Import it to our projects directly if that's a good idea.

Comment (mwiencek) [reply scss/bootstrap-prod.scss]: > Include this before custom-vars works fine btw!

That doesn't appear to be strictly true; check the diff for dist/bootstrap.css where e.g. the colors defined at the top are wrong now, whereas they were correct before.

File: dist/bootstrap.css [hunk 7]
```diff
@@ -4,83 +4,155 @@
...
+.bs {
+  /* rtl:raw:
+  [type="tel"],
+  [type="url"],
+  [type="email"],
+  [type="number"] {
+    direction: ltr;
+  }
+  */
+  @import "https://fonts.googleapis.com/css?family=Comfortaa:300,400,700";
```
Comment (mwiencek) [inline dist/bootstrap.css]: Any idea why this is this importing a font from Google servers now?  We need to prevent this.

File: package.json (hunk 4 above)
Comment (mwiencek) [reply package.json:45]: Thanks, but it should be added to the commit message too. :)

File: components/index.js (hunk 5 above)
Comment (mwiencek) [reply components/index.js]: You can remove it from the commit with `git rebase -i ...` (where ... is the commit prior to the one you want to start editing at).

**Comment by mwiencek** [comment]
This PR still has Header and Footer components but as mentioned above they are very MB-specific, use translated strings, etc., so don't they belong in musicbrainz-server?

I'd greatly appreciate if we can clean up the commits in this PR so that we're not adding things and then reverting them (as is the case with the css minimization, typescript, etc.).  Here's a nice guide on interactive rebasing: https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history

**Comment by akshaaatt** [comment]
Will work on a smaller PR to get this work done.
